src/connector/src/error.rs

```diff
@@ -41,17 +42,21 @@ def_anyhow_newtype! {
     url::ParseError => "failed to parse url",
     serde_json::Error => "failed to parse json",
     csv::Error => "failed to parse csv",
+    ArrayError => transparent,
+
     uuid::Error => transparent, // believed to be self-explanatory
 
     // Connector errors
     opendal::Error => transparent, // believed to be self-explanatory
+    parquet::errors::ParquetError => transparent,
```

Comment:
Have you checked they are self-explanatory? If not, please add additional error message.